kgforge/core/archetypes/store.py

```diff
@@ -443,11 +303,9 @@ def sparql(
             self._debug_query(qr)
         return self._sparql(qr)
 
-    def _sparql(self, query: str) -> List[Resource]:
-        # POLICY Should notify of failures with exception QueryingError including a message.
-        # POLICY Resource _store_metadata should not be set (default is None).
-        # POLICY Resource _synchronized should not be set (default is False).
-        not_supported()
+    @abstractmethod
+    def _sparql():
```

Comment:
should have a self argument, and a query: str argument if you expect that this _sparql should be called by specifying only one str parameter (which is how it's called in sparql(..) )

Reply:
I guess it should be the way it was before except it's now abstract